src/uu/df/src/df.rs

```diff
@@ -155,8 +162,12 @@ impl Options {
         Ok(Self {
             show_local_fs: matches.is_present(OPT_LOCAL),
             show_all_fs: matches.is_present(OPT_ALL),
-            block_size: block_size_from_matches(matches)
-                .map_err(|_| OptionsError::InvalidBlockSize)?,
+            block_size: block_size_from_matches(matches).map_err(|e| match e {
+                ParseSizeError::SizeTooBig(_) => OptionsError::BlockSizeTooLarge(
+                    matches.value_of(OPT_BLOCKSIZE).unwrap().to_string(),
+                ),
+                ParseSizeError::ParseFailure(_) => OptionsError::InvalidBlockSize,
```

Comment:
It would be helpful to add a test for this case as well.

Reply:
Ok, I added a test and changed `OptionsError::InvalidBlockSize` to `OptionsError::InvalidBlockSize(String)`.